server/py/framework/db/sqldb/db.py

```diff
@@ -818,6 +819,27 @@ def create_artifact(
 
         return uid
 
+    def _get_parent_artifact_id(self, session, parent_uri: str) -> int:
+        if parent_uri:
+            _, uri = mlrun.datastore.parse_store_uri(parent_uri)
+            project, key, iteration, tag, tree, uid = parse_artifact_uri(uri)
+            parent_db_artifact = self.read_artifact(
+                session=session,
+                key=key,
+                tag=tag,
+                iter=iteration,
+                producer_id=tree,
+                uid=uid,
+                project=project,
```

Comment:
hmm... This made me think - we don't block cross-project reference here, and this is potentially a security breach where a prompt artifact actually points at a model from another project. This retrieval here is done without any checking that the user is allowed to access the project pointed at by the uri.
For now I think you need to block cross-project references (in the sdk and in the server side), we need to consider what it means to allow cross-project references, and whether it makes sense.